src/Lecture2.hs

```diff
@@ -48,7 +49,13 @@ zero, you can stop calculating product and return 0 immediately.
 84
 -}
 lazyProduct :: [Int] -> Int
-lazyProduct = error "TODO"
+lazyProduct = go 1
+  where
+    go :: Int -> [Int] -> Int
+    go acc [] = acc
+    go acc (x : xs)
+      | x == 0    = 0
```

Comment:
Correct solution 👍🏻 Alternatively, you can match on zero in the pattern as well

```suggestion
    go acc (0 : _) = 0
```

Reply:
updated